rzup/src/error.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2025 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use thiserror::Error;
+
+#[derive(Debug, Error, Clone)]
+pub enum RzupError {
+    #[error("Component not found: {0}")]
+    ComponentNotFound(String),
+
+    #[error("Installation failed: {0}")]
+    InstallationFailed(String),
+
+    #[error("Invalid version: {0}")]
+    InvalidVersion(String),
+
+    #[error("IO error: {0}")]
+    Io(String),
+
+    #[error("Environment error: {0}")]
+    Environment(String),
+
+    #[error("{0}")]
+    Other(String),
+
+    #[error("Rate Limited: {0}")]
+    RateLimited(String),
+}
+
+impl From<std::io::Error> for RzupError {
+    fn from(e: std::io::Error) -> Self {
+        RzupError::Io(e.to_string())
+    }
+}
+
+impl From<anyhow::Error> for RzupError {
+    fn from(e: anyhow::Error) -> Self {
+        RzupError::Other(e.to_string())
```

Comment:
Maybe it would be worth trying to downcast this error to see if it is an `std::io::Error` and using `RzUp::Io` variant in that case?